api/v1beta1/azuremanagedcontrolplane_webhook_test.go

```diff
@@ -2683,3 +2770,670 @@ func getAMCPMetaData() metav1.ObjectMeta {
+			errLen:  1,
+		},
+		{
+			name: "When AzureKms.KeyVaultNetworkAccess is public AzureKeyVaultKms.KeyVaultResourceID should be empty",
+			amcp: &AzureManagedControlPlane{
+				Spec: AzureManagedControlPlaneSpec{
+					AzureManagedControlPlaneClassSpec: AzureManagedControlPlaneClassSpec{
+						Version: "v1.17.8",
+						Identity: &Identity{
+							Type:                           ManagedControlPlaneIdentityTypeUserAssigned,
+							UserAssignedIdentityResourceID: "not empty",
+						},
+					},
+					SecurityProfile: &ManagedClusterSecurityProfile{
+						AzureKeyVaultKms: &AzureKeyVaultKms{
+							Enabled:               true,
+							KeyID:                 "not empty",
+							KeyVaultNetworkAccess: ptr.To(KeyVaultNetworkAccessTypesPublic),
+							KeyVaultResourceID:    ptr.To("not empty"),
+						},
+					},
+				},
+			},
+			wantErr: true,
+			errLen:  1,
+		},
+		{
+			name: "Image Cleaner Interval hours has to be in range [24-2160]",
+			amcp: &AzureManagedControlPlane{
+				Spec: AzureManagedControlPlaneSpec{
+					AzureManagedControlPlaneClassSpec: AzureManagedControlPlaneClassSpec{
+						Version: "v1.17.8",
+					},
+					SecurityProfile: &ManagedClusterSecurityProfile{
+						ImageCleaner: &ManagedClusterSecurityProfileImageCleaner{
+							Enabled:       true,
+							IntervalHours: ptr.To(0),
+						},
+					},
+				},
+			},
+			wantErr: true,
+			errLen:  1,
+		},
+		{
+			name: "Valid profile",
+			amcp: &AzureManagedControlPlane{
+				Spec: AzureManagedControlPlaneSpec{
+					AzureManagedControlPlaneClassSpec: AzureManagedControlPlaneClassSpec{
+						Version: "v1.17.8",
+						Identity: &Identity{
+							Type:                           ManagedControlPlaneIdentityTypeUserAssigned,
+							UserAssignedIdentityResourceID: "not empty",
+						},
+						OIDCIssuerProfile: &OIDCIssuerProfile{
+							Enabled: ptr.To(true),
+						},
+					},
+					SecurityProfile: &ManagedClusterSecurityProfile{
+						AzureKeyVaultKms: &AzureKeyVaultKms{
+							Enabled:               true,
+							KeyID:                 "not empty",
+							KeyVaultNetworkAccess: ptr.To(KeyVaultNetworkAccessTypesPublic),
+						},
+						Defender: &ManagedClusterSecurityProfileDefender{
+							LogAnalyticsWorkspaceResourceID: "not empty",
+							SecurityMonitoring: ManagedClusterSecurityProfileDefenderSecurityMonitoring{
+								Enabled: true,
+							},
+						},
+						WorkloadIdentity: &ManagedClusterSecurityProfileWorkloadIdentity{
+							Enabled: true,
+						},
+						ImageCleaner: &ManagedClusterSecurityProfileImageCleaner{
+							Enabled:       true,
+							IntervalHours: ptr.To(24),
+						},
+					},
+				},
+			},
+			wantErr: false,
+			errLen:  0,
+		},
+	}
+	client := mockClient{ReturnError: false}
+	for _, tc := range testsCreate {
+		t.Run(tc.name, func(t *testing.T) {
+			g := NewWithT(t)
+			mcpw := &azureManagedControlPlaneWebhook{
+				Client: client,
+			}
+			_, err := mcpw.ValidateCreate(context.Background(), tc.amcp)
```

Comment:
Out of curiosity, is there a reason these tests need their own top-level function, or could these cases fit into an existing table?

Reply:
Did not want to crowd the above function, as it is already too big.